plugin/storage/grpc/shared/streaming_writer_grpc_client.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2022 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shared
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/proto-gen/storage_v1"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+)
+
+type streamingWriterGRPCClient struct {
+	*grpcClient
+	streamingWriterClient storage_v1.StreamingSpanWriterPluginClient
```

Comment:
```suggestion
	client storage_v1.StreamingSpanWriterPluginClient
```

Reply:
@yurishkuro Thanks for the feedback.

> I feel like I am missing something. It's not clear to me why we need a separate "streaming client" writer, could we not do it from the existing client?

We can do it in the existing client, I want to write the code more elegantly, if there is any suggestion please do not hesitate to teach in the next review.

> But more importantly, I think the overall approach we had taken when we extended the plugin contract with "archive" storage was to add Capabilities field that tells the client if the plugin implementation supports additional interfaces. So I am surprised that you don't have any changed to the Factory files.

ok, I'll change to using Capabilities to negotiate.

> Lastly, I would like to have integration tests to exercise the new functionality.

ok, I'll add it